cmd/nerdctl/container/container_commit_linux_test.go

```diff
@@ -21,76 +21,70 @@ import (
-Unfortunately, this is flaky.
-It will regularly succeed or fail, making random PR fail the Kube check.
-*/
-
-func TestKubeCommitPush(t *testing.T) {
-	t.Parallel()
-
-	base := testutil.NewBaseForKubernetes(t)
-	tID := testutil.Identifier(t)
-
-	var containerID string
-	// var registryIP string
-
-	setup := func() {
-		testutil.KubectlHelper(base, "run", "--image", testutil.CommonImage, tID, "--", "sleep", "Inf").
-			AssertOK()
-
-		testutil.KubectlHelper(base, "wait", "pod", tID, "--for=condition=ready", "--timeout=1m").
-			AssertOK()
-
-		testutil.KubectlHelper(base, "exec", tID, "--", "mkdir", "-p", "/tmp/whatever").
-			AssertOK()
-
-		cmd := testutil.KubectlHelper(base, "get", "pods", tID, "-o", "jsonpath={ .status.containerStatuses[0].containerID }")
-		cmd.Run()
-		containerID = strings.TrimPrefix(cmd.Out(), "containerd://")
-
-		// This below is missing configuration to allow for plain http communication
-		// This is left here for future work to successfully start a registry usable in the cluster
-		/*
-			// Start a registry
-					testutil.KubectlHelper(base, "run", "--port", "5000", "--image", testutil.RegistryImageStable, "testregistry").
-						AssertOK()
-
-					testutil.KubectlHelper(base, "wait", "pod", "testregistry", "--for=condition=ready", "--timeout=1m").
-						AssertOK()
-
-					cmd = testutil.KubectlHelper(base, "get", "pods", tID, "-o", "jsonpath={ .status.hostIPs[0].ip }")
-					cmd.Run()
-					registryIP = cmd.Out()
-
-					cmd = testutil.KubectlHelper(base, "apply", "-f", "-", fmt.Sprintf(`apiVersion: v1
-				kind: ConfigMap
-				metadata:
-					name: local-registry
-					namespace: nerdctl-test
-				data:
-					localRegistryHosting.v1: |
-					host: "%s:5000"
-					help: "https://kind.sigs.k8s.io/docs/user/local-registry/"
-			`, registryIP))
-		*/
-
+func TestKubeCommitSave(t *testing.T) {
+	testCase := nerdtest.Setup()
+
+	testCase.Require = nerdtest.OnlyKubernetes
+
+	testCase.Setup = func(data test.Data, helpers test.Helpers) {
+		containerID := ""
+		// NOTE: kubectl namespaces are not the same as containerd namespaces.
+		// We still want kube test objects segregated in their own Kube API namespace.
+		nerdtest.KubeCtlCommand(helpers, "create", "namespace", "nerdctl-test-k8s").Run(&test.Expected{})
+		nerdtest.KubeCtlCommand(helpers, "run", "--image", testutil.CommonImage, data.Identifier(), "--", "sleep", "Inf").Run(&test.Expected{})
+		nerdtest.KubeCtlCommand(helpers, "wait", "pod", data.Identifier(), "--for=condition=ready", "--timeout=1m").Run(&test.Expected{})
+		nerdtest.KubeCtlCommand(helpers, "exec", data.Identifier(), "--", "mkdir", "-p", "/tmp/whatever").Run(&test.Expected{})
+		nerdtest.KubeCtlCommand(helpers, "get", "pods", data.Identifier(), "-o", "jsonpath={ .status.containerStatuses[0].containerID }").Run(&test.Expected{
+			Output: func(stdout string, info string, t *testing.T) {
+				containerID = strings.TrimPrefix(stdout, "containerd://")
+			},
+		})
```

Comment:
Not necessary per-se, but what asked for when the k8s env was introduced: https://github.com/containerd/nerdctl/pull/3296#discussion_r1719194632

Reply:
> Not necessary per-se

will these commands be run in the default namespace or in `nerdctl-test-k8s`?

Reply:
k8s namespaces are not the same as containerd namespaces (yeah, really confusing).

Commands run by `nerdctl` for _this environment_ are run inside the `k8s.io` containerd namespace (instead of the `nerdctl-test` usual namespace).

Then `nerdctl-test-k8s` is a k8 namespace, unrelated.

Also note that these are running against the containerd provided by kind, which is inside a container IIRC, and not on the host.